pkg/apis/apis/v1alpha1/types_apiexport.go

```diff
@@ -220,24 +220,34 @@ type PermissionClaim struct {
 	IdentityHash string `json:"identityHash,omitempty"`
 }
 
-// +kubebuilder:validation:XValidation:rule="has(self.__namespace__) || has(self.name)",message="at least one field must be set"
+const (
+	ResourceSelectorAll = "*"
+)
+
+// ResourceSelector identifies the objects to be included within a PermissionClaim either by name and/or namespace.
+//
+// +kubebuilder:validation:XValidation:rule="has(self.namespaces) || has(self.names)",message="at least one field must be set"
 type ResourceSelector struct {
-	// name of an object within a claimed group/resource.
+	// names is a list of specific resources to select.
 	// It matches the metadata.name field of the underlying object.
-	// If namespace is unset, all objects matching that name will be claimed.
+	// An entry of "*" anywhere in the list means all object names of the group/resource within the "namespaces" list are claimed.
 	//
 	// +optional
-	// +kubebuilder:validation:Pattern="^([a-z0-9][-a-z0-9_.]*)?[a-z0-9]$"
-	// +kubebuilder:validation:MaxLength=253
-	// +kubebuilder:validation:MinLength=1
-	Name string `json:"name,omitempty"`
+	// +listType=set
+	// +kubebuilder:default={"*"}
+	Names []string `json:"names,omitempty"`
```

Comment:
specify what empty list means? is it defaulted to `["*"]` or not?

Reply:
What does "**" mean? What does "foo*" mean?

Reply:
The comments start at L231. We're missing `foo*` for sure.

Reply:
For this PR, I'd like to leave wildcard strings out and iterate further.